postreise/plot/plot_bar_generation_stack.py

```diff
@@ -0,0 +1,234 @@
+    if not scenario_ids:
+        scenario_ids = []
+    if isinstance(scenario_ids, (int, str)):
+        scenario_ids = [scenario_ids]
+    if isinstance(resources, str):
+        resources = [resources]
+    if not isinstance(resources, list):
+        raise TypeError("ERROR: resources should be a list")
+    if isinstance(area_types, str):
+        area_types = [area_types]
+    if not area_types:
+        area_types = [None] * len(areas)
+    if len(areas) != len(area_types):
+        raise ValueError(
+            "ERROR: if area_types are provided, it should have the same number of entries with areas"
+        )
+    if isinstance(scenario_names, str):
+        scenario_names = [scenario_names]
+    if scenario_names and len(scenario_names) != len(scenario_ids):
+        raise ValueError(
+            "ERROR: if scenario names are provided, number of scenario names must match number of scenario ids"
+        )
+    if titles is not None and not isinstance(titles, dict):
+        raise TypeError("ERROR: titles should be a dictionary if provided")
+    if filenames is not None and not isinstance(filenames, dict):
+        raise TypeError("ERROR: filenames should be a dictionary if provided")
+    s_list = []
+    for sid in scenario_ids:
+        s_list.append(Scenario(sid))
+    mi = ModelImmutables(s_list[0].info["grid_model"])
+    type2color = mi.plants["type2color"]
+    type2label = mi.plants["type2label"]
+    type2hatchcolor = mi.plants["type2hatchcolor"]
+    if t2c:
+        type2color.update(t2c)
+    if t2l:
+        type2label.update(t2l)
+    if t2hc:
+        type2hatchcolor.update(t2hc)
+    all_loadzone_data = dict()
+    for sid, scenario in zip(scenario_ids, s_list):
+        curtailment = calculate_curtailment_time_series_by_areas_and_resources(
+            scenario, areas={"loadzone": mi.zones["loadzone"]}
+        )
+        for area in curtailment:
+            for r in curtailment[area]:
+                curtailment[area][r] = curtailment[area][r].sum().sum()
+        curtailment = (
+            pd.DataFrame(curtailment).rename(columns=mi.zones["loadzone2id"]).T
+        )
+        curtailment.rename(
+            columns={c: c + "_curtailment" for c in curtailment.columns}, inplace=True
+        )
+        curtailment["curtailment"] = curtailment.sum(axis=1)
+        all_loadzone_data[sid] = pd.concat(
+            [
+                sum_generation_by_type_zone(scenario).T,
+                scenario.state.get_demand().sum().T.rename("load"),
+                curtailment,
+            ],
+            axis=1,
+        ).rename(index=mi.zones["id2loadzone"])
+
+    width = 0.4
+    x_scale = 0.6
+    ax_list = []
+    for area, area_type in zip(areas, area_types):
+        fig, ax = plt.subplots(figsize=(10, 8))
+        for ind, s in enumerate(s_list):
+            patches = []
+            fuels = []
+            bottom = 0
+            loadzone_set = area_to_loadzone(s.info["grid_model"], area, area_type)
```

Comment:
I guess that it will reak here if we have scenarios that have different grid model since an area might be defined in one grid model but not in another, e.g, Bay Area is a load zone in usa_tamu but not in hifld.

Reply:
> I guess that it will break here if we have scenarios that have different grid model since an area might be defined in one grid model but not in another, e.g, Bay Area is a load zone in usa_tamu but not in hifld.

Yeah, I think it should be fine to break here by printing out "invalid area" from `area_to_loadzone` function to let the user know a specific area is not available for a particular `grid_model`?